### PR TITLE
Add oniguruma package

### PR DESCRIPTION
**Pull request by isaachier**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Yes**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/isaachier/hunter/build/1.0.428
  * https://travis-ci.org/isaachier/hunter/builds/358912772

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/214

### CODE REVIEW

File: docs/packages/pkg/oniguruma.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. spelling::
+
+    oniguruma
+
+.. index:: regex ; oniguruma
+
+.. _pkg.oniguruma:
+
+oniguruma
+===
```
Comment (ruslo) [inline docs/packages/pkg/oniguruma.rst]: Need more `=` here.

File: docs/packages/regex.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.. spelling::
+
+  libpcre
+  oniguruma
+
+Regex
+-----
+
+ - :ref:`pkg.libpcre` - Perl-compatible regular expression library
+ - :ref:`pkg.oniguruma` - modern and flexible regular expressions library
```
Comment (ruslo) [inline docs/packages/regex.rst]: > regular expressions library
> regular expression library

Not consistent

Comment (isaachier) [reply docs/packages/regex.rst]: Good point

File: docs/packages/pkg/oniguruma.rst (hunk 1 above)
Comment (isaachier) [reply docs/packages/pkg/oniguruma.rst]: Missed that thanks

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.3540
* https://travis-ci.org/ingenue/hunter/builds/359028156?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.3544
* https://travis-ci.org/ingenue/hunter/builds/359350412?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.20.31

**Pull request merged**